x-pack/plugins/cloud_security_posture/public/pages/compliance_dashboard/compliance_dashboard.test.tsx

```diff
@@ -387,6 +392,8 @@ describe('<ComplianceDashboard />', () => {
   });
 
   it('Show Kubernetes dashboard if there are KSPM findings', () => {
+    (useLocalStorage as jest.Mock).mockReturnValue([KSPM_POLICY_TEMPLATE, jest.fn()]);
```

Comment:
please revert this line - it breaks the test's requirement

Reply:
@kfirpeled  do you mind explaining more how the changes break the test requirement?
If the dashboard has kspm findings, then we set the  lastSelectedTab in local storage with KSPM?

Reply:
the requirement is, that when there's no data in the localstorage, the default selection is by CSPM first, if there are CSPM findings. And KSPM, if there are KSPM findings. When there are no findings, it picks CSPM onboarding.

but if there's localstorage, it should pick the local storage. And by reading the test definition: `Show Kubernetes dashboard if there are KSPM findings'` I can see you've altered that condition

Reply:
* No data in localstorage is cspm by default
* If there is cspm findings, but no kspm findings,, then tab is cspm and set to localstorage  tab to cspm
* If there is kspm findings but not cspm findings, set tab to kspm and set to localstorage tab to kspm
* if there is both kspm and cspm findings, then set tab to CSPM and set to localstorage  tab to  cspm
I added the localstorage line for Arrange section of the test since we are setting  the localstorage  tab to KSPM.  If 
`  const [selectedTabLocalStorage, setSelectedTabLocalStorage] =
    useLocalStorage<PosturePolicyTemplate>(LOCAL_STORAGE_COMPLIANCE_DASHBOARD_SELECTED_TAB_KEY);`

I could be wrong. If we are using the `setSelectedTabLocalStorage` and `selectedTabLocalStorage`  as dependencies ` below then shouldn't local storage setup be an _Arrange_ setup pre-condition to meet `Show Kubernetes dashboard if there are KSPM findings` requirement.

  ```useEffect(() => {
    if (selectedTabLocalStorage) {
      return;
    }

    const preferredDashboard = getDefaultTab(
      getSetupStatus,
      getCspmDashboardData.data,
      getKspmDashboardData.data
    );
    setSelectedTabLocalStorage(preferredDashboard);
  }, [
    getCspmDashboardData.data,
    selectedTabLocalStorage,
    setSelectedTabLocalStorage,
    getCspmDashboardData.data?.stats.totalFindings,
    getKspmDashboardData.data,
    getKspmDashboardData.data?.stats.totalFindings,
    getSetupStatus,
    getSetupStatus?.cspm?.status,
    getSetupStatus?.kspm?.status,
  ]);
```
and if i revert the line then test would break?

Reply:
I'm not sure why `getDefaultTab` also sets the local storage, what's the rational here?